### PR TITLE
Fake mappings should be merged into a single one during merging

### PR DESCRIPTION
**Pull request by jqll**

### CODE REVIEW

File: profile/merge_test.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
```
Comment (aalexand) [inline profile/merge_test.go]: Nit: fix the year.

File: profile/merge.go [hunk 2]
```diff
@@ -324,41 +315,38 @@ func (pm *profileMerger) mapMapping(src *Mapping) mapInfo {
...
+		offset: m.Offset,
 	}
 
-	pathKey = mappingKey{
-		size,
-		m.Offset,
-		m.File,
+	switch {
+	case m.BuildID != "":
+		key.buildidIDOrFile = m.BuildID
```
[the remark this answers is not in the document]
Comment (jqll) [reply profile/merge.go]: Fixed. Not sure if buildidIDOrFile was intentional, but buildIDOrFile sounds enough.

File: profile/merge_test.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package profile
+
+import (
+	"testing"
+)
+
+func TestMapMapping(t *testing.T) {
```
Comment (aalexand) [inline profile/merge_test.go]: Please confirm the test fails before the fix.

**Comment by aalexand** [comment]
I'll merge once CI finishes.

File: profile/merge_test.go (hunk 3 above)
Comment (jqll) [reply profile/merge_test.go]: Yes, TestMapMapping/same_fake_mapping fails before the change.

**Comment by aalexand** [comment]
@jqll CI fails, PTAL.

**Pull request merged**